Jetpack Forms: Add tracking of Google Sheets exports of responses

donnchawp opened the pull request:
This is a dupe of #29119.

When responses are exported from WP-Admin->Feedback->Form Responses they can be exported to Google Sheets. This patch adds tracking of those actions.

It requires the record_tracks_event function from https://github.com/Automattic/jetpack/pull/29102.

Discussion: pcsBup-S8-p2
Testing instructions:

Wait for https://github.com/Automattic/jetpack/pull/29102 to be merged or grab record_tracks_event from it.
Create a form block in a post with /form.
Fill out the form and submit.
Go to WP-Admin->Feedback->Form Responses and export the response(s) to Google Sheets. You'll have to connect it first. If using Firefox containers you might have to do this bit in Chrome.
If an Automattician, look for the forms_export_responses tracking event. The format prop should be "gsheets"

## Does this pull request change what data or activity we track or use?

Yes, it tracks exports by Google Sheets from the Feedback Forms page.

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

donnchawp committed:
Add tracking of Google Sheets exports

donnchawp committed:
Changelog

donnchawp committed:
changelog

donnchawp committed:
Update add-tracking_of_gsheets_export#2

CGastrell reviewed (dismissed):
Still looks good to me :shipit:

anomiex committed:
[not verified] cli: `generate` command should ask for plugin versioning method (#29078)

* cli: `generate` command should ask for plugin versioning method

We've been defaulting to semver like we do for all other project types,
but it seems this may be confusing teams working on the plugins who may
be expecting WordPress-style versioning. Instead let's ask when the
plugin is being created.

samiff committed:
[not verified] jetpack-mu-wpcom: fix undefined is_plugin_active fatal on wpcom (#29158)

jeyip committed:
[not verified] Launchpad: Refactor launchpad save modal (#29134)

ice9js committed:
[not verified] Add page navigation for the new feedback dashboard (#28826)

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

IanRamosC committed:
[not verified] Fix/discount for first month trials (#29139)

* Add a pricing-utils helper folder

This folder will store helper functions related to pricing. For now, I've only included `isFirstMonthTrial` as it's the only one required for this PR, but intention is to add more functions related to pricing, instead of having them only as reducers/selectors.

* Recommendations: remove discount if there's a trial

* Recommendations: remove the coupon code from the URL if there's a trial

This change is made in the component instead of in withUpgradeUrl because this rule is more strict and related specifically to what is displayed in this card. This change is mostly to avoid visual confusion.

* Add changelogger

* Remove unnecessary CSS

* Fix changelogger validity

ouikhuan committed:
[not verified] Replace links with Jetpack redirect service for migration plugin (#29129)

jcheringer committed:
[not verified] Forms: Increase form fields padding based on user-defined border-radius (#28820)

* Increase form fields padding based on user-defined border-radius

* changelog

* Fix field padding in the Forms package

* changelog

* Fix changelogger

a8ck3n committed:
[not verified] Odyssey: Remove Calypso recommendations from legacy stats page (#29115)

* Odyssey: Remove Calypso nudge.

Doesn’t make sense to push to Calypso now that Odyssey is an option.

* Odyssey: Remove Calypso nudge (2).

This time from the loading UI.

* Changelog++

oskosk committed:
[not verified] Related Posts: Add block support in server block renderer (#29135)

* [not verified] Add block support for font family in Related Posts block server block renderer

* [not verified] changelog

retrofox committed:
[not verified]  VideoPress: remove uploading image for video poster when uploading video (#29183)

* remove image for poster when uploading video

* fix default aspect ratio

* changelog

jeherve committed:
[not verified] Widget Visibility: switch to shared Analytics implementation. (#29181)

samiff committed:
[not verified] Jetpack: update to-test.md for v11.9 (#29186)

* Jetpack: update to-test.md for v11.9

* Typos and formatting

anomiex committed:
[not verified] build: Cleanup workflows (#29188)

nateweller committed:
[not verified] WAF: Add support for handling IP ranges in allow/block lists (#29131)

ice9js committed:
[not verified] Make Jetpack Forms dashboard use the "Feedback" menu item in WP Admin (#29198)

* Update the forms dashboard to use the same feedback menu item

* changelog

gmjuhasz committed:
[not verified] Remove `Default Image` option from SIG panel (#29206)

* Remove default image type as it's not used yet

* changelog

jeherve committed:
[not verified] Custom CSS: use slug to link to Custom CSS settings (#29202)

jeherve committed:
[not verified] Blocks: update scaffolding CLI commands (#29201)

danielbachhuber committed:
[not verified] Staging Sites: Add staging site attributes to the site object (#29192)

* REST API: Add staging site attributes to the site object

* Make the key name more specific

donnchawp commented:
Abandoning this PR too as rebase went astray.
